ldm/simplet2i.py

```diff
@@ -534,6 +551,55 @@ def _new_seed(self):
         self.seed = random.randrange(0, np.iinfo(np.uint32).max)
         return self.seed
 
+    def _get_variation_noise(self, width:int, height:int, variant_amount:float, variant_seed:int) -> "tuple[torch.Tensor,torch.Tensor]":
+        base_x_T = None
+        target_x_T = None
+        if variant_amount is not None:
+            variant_amount = max(0.0, min(1.0, variant_amount))
+            # base noise is made from our initial seed or seed provided with -S
+            base_x_T = torch.randn([self.batch_size,
+                                    self.latent_channels,
+                                    height // self.downsampling_factor,
+                                    width  // self.downsampling_factor],
+                                    device=self.device)
+            if variant_seed is not None:
+                # store initial seed
+                initialSeed = torch.initial_seed()
+                # generate target noise from the provided variant seed
+                seed_everything(variant_seed)
+                target_x_T = torch.randn([self.batch_size,
+                                self.latent_channels,
+                                height // self.downsampling_factor,
+                                width  // self.downsampling_factor],
+                                device=self.device)
+                # switch back to the initial seed
+                seed_everything(initialSeed)
+        return base_x_T, target_x_T
+
+    def _apply_variation_slerp(self, 
+        width:int, height:int, steps:int, 
+        variant_amount:float, variant_seed:int, 
+        base_x_T:torch.Tensor, target_x_T:torch.Tensor) -> torch.Tensor:
+        x_T = None
+        if variant_amount is not None:
+            variant_amount = max(0.0, min(1.0, variant_amount))
+            # no variant seed specified, generate random noise
+            if variant_seed is None:
+                target_x_T = torch.randn([self.batch_size,
+                                self.latent_channels,
+                                height // self.downsampling_factor,
+                                width  // self.downsampling_factor],
+                                device=self.device)
+
+            # slerp base -> target using variant amount
+            x_T = self.slerp(variant_amount, base_x_T, target_x_T)
+
+            # only for ksampler!
+            if isinstance(self.sampler, KSampler):
+                # KSampler does not do it when x_T provided
```

Comment:
Maybe instead change `KSampler` to include this? Seems best not to need a special case here, and if the other samplers work without this, KSampler ought to as well.

Reply:
sounds good, I removed the special case, I think it is needed in KSampler for img2img also

Reply:
> sounds good, I removed the special case, I think it is needed in KSampler for img2img also

Hmm. We should implement the remaining samplers for img2img.